internal/state/graph.go

```diff
@@ -47,35 +56,80 @@ type gatewayClass struct {
 }
 
 // graph is a graph-like representation of Gateway API resources.
-// It is assumed that the Gateway resource is a single resource.
 type graph struct {
 	// GatewayClass holds the GatewayClass resource.
 	GatewayClass *gatewayClass
-	// Listeners holds listeners keyed by their names in the Gateway resource.
-	Listeners map[string]*listener
+	// Gateway holds the winning Gateway resource.
+	Gateway *gateway
```

Comment:
The types are private but the fields are public. How is this intended to be used?

Within the package the struct fields, public or private, should be accessible. Will graph and its dependents only be used in `package state`?

Reply:
the `graph` and its dependets types are only meant to be used in `package state`